zulip_bots/zulip_bots/bots/weather/weather.py

```diff
@@ -9,6 +10,21 @@ def initialize(self, bot_handler: Any) -> None:
         self.api_key = bot_handler.get_config_info('weather')['key']
         self.response_pattern = 'Weather in {}, {}:\n{:.2f} F / {:.2f} C\n{}'
 
+        self.check_api_key()
+
+    def check_api_key(self) -> None:
+        url = 'http://api.openweathermap.org/data/2.5/weather?q=nyc&APPID=' + self.api_key
+        test_response = requests.get(url)
+
+        try:
+            test_response_data = test_response.json()
+
+            error_response = 'Invalid API key. Please see http://openweathermap.org/faq#error401 for more info.'
+            if test_response_data['cod'] == 401 and test_response_data['message'] == error_response:
```

Comment:
The check for the 401 suffices for checking for invalid credentials. I'll remove the check for the error response, since that string might change independently from the error type.